mixer/pkg/server/server.go

```diff
@@ -205,10 +201,18 @@ func newServer(a *Args, p *patchTable) (*Server, error) {
 	rt = p.newRuntime(st, templateMap, adapterMap, a.ConfigDefaultNamespace,
 		s.gp, s.adapterGP, a.TracingOptions.TracingEnabled())
 
+	// this method initializes the store
 	if err = p.runtimeListen(rt); err != nil {
 		_ = s.Close()
 		return nil, fmt.Errorf("unable to listen: %v", err)
 	}
+
+	// block wait for the config store to sync
+	log.Info("Awaiting for config store sync...")
+	if err := st.WaitForSynced(30 * time.Second); err != nil {
```

Comment:
I think it would be safer to just wait - and let the readiness probe take care of killing the process. It's more configurable than hardcoding 30 seconds - and exiting after ( I assume return err results in mixer restarting )

Reply:
That is tying this too closely to k8s / readiness.
This will show a more specific message 
```
Awaiting for config store sync...
exceeded timeout 30s
```
@kyessenov can you confirm that the resulting error is correctly printed?

Reply:
With timeout 0:
```
2018-08-16T20:57:16.013036Z     info    Awaiting for config store sync...
Unable to initialize Mixer: exceeded timeout 0s
exit status 255
echo $?
1
```